### PR TITLE
Check for omnibus pinned commit presence in their respective branches

### PR DESCRIPTION
**Pull request by chouquette**
### What does this PR do?

This PR prepares our do-not-merge check for omnibus pinned commits.
We currently check that the targeted branch is `master`/`datadog-5.5.0` or a stable branch, which won't work when we start pinning specific commits.

### Motivation

Have green CI when we start pinning omnibus commits

### Additional Notes

Passing job: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/404048498
Failing job: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/404041493

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by chouquette** [comment]
Manually tested on `7.50.x`:
```
$> inv -e release.check-omnibus-branches
git clone --depth=50 https://github.com/datadog/omnibus-ruby --branch 7.50.x /tmp/tmph3683mze/omnibus-ruby
Cloning into '/tmp/tmph3683mze/omnibus-ruby'...
git clone --depth=50 https://github.com/datadog/omnibus-software --branch 7.50.x /tmp/tmph3683mze/omnibus-software
Cloning into '/tmp/tmph3683mze/omnibus-software'...
omnibus-software commit (8f416d7e9071d50ae8410b058d307556e38a51cd) is not in the expected branch (7.50.x). The PR is not mergeable
```

```
inv -e release.check-omnibus-branches
git clone --depth=50 https://github.com/datadog/omnibus-ruby --branch 7.50.x /tmp/tmp9__6fi6x/omnibus-ruby
Cloning into '/tmp/tmp9__6fi6x/omnibus-ruby'...
git clone --depth=50 https://github.com/datadog/omnibus-software --branch 7.50.x /tmp/tmp9__6fi6x/omnibus-software
Cloning into '/tmp/tmp9__6fi6x/omnibus-software'...
omnibus-ruby commit (f3cbfdcf00a208cc165a30da80a95c456b5fa25b) is not in the expected branch (7.50.x). The PR is not mergeable
```

File: release.json [hunk 1]
```diff
@@ -26,8 +26,8 @@
     },
     "nightly-a7": {
         "INTEGRATIONS_CORE_VERSION": "master",
-        "OMNIBUS_SOFTWARE_VERSION": "master",
-        "OMNIBUS_RUBY_VERSION": "datadog-5.5.0",
+        "OMNIBUS_SOFTWARE_VERSION": "439c4abf113b68702b63d9c46bbcc24a176eb808",
+        "OMNIBUS_RUBY_VERSION": "982a483f9cc36f0ee43928e03c0e8db3902ac9bc",
```
Comment (richl-dd) [inline release.json:30]: Same as https://github.com/DataDog/datadog-agent/pull/21961/files#r1446383836

**Review by richl-dd — APPROVED** [review]
Looks good - could you double check the comments on the changes in `release.json` regarding the versions prior to merge?

File: release.json [hunk 2]
```diff
@@ -6,8 +6,8 @@
     },
     "nightly": {
         "INTEGRATIONS_CORE_VERSION": "master",
-        "OMNIBUS_SOFTWARE_VERSION": "master",
-        "OMNIBUS_RUBY_VERSION": "datadog-5.5.0",
+        "OMNIBUS_SOFTWARE_VERSION": "439c4abf113b68702b63d9c46bbcc24a176eb808",
+        "OMNIBUS_RUBY_VERSION": "982a483f9cc36f0ee43928e03c0e8db3902ac9bc",
```
[the remark this answers is not in the document]
Comment (chouquette) [reply release.json:10]: This was used for testing, but I need to revert it indeed. Thanks!

File: tasks/release.py [hunk 3]
```diff
@@ -1412,15 +1413,55 @@ def cleanup(ctx):
...
+                    f'git -C {tmpdir}/omnibus-ruby branch --contains {omnibus_ruby_commit}', warn=True, hide=True
+                ).exited
+                != 0
+            ):
+                raise Exit(
+                    code=1,
+                    message=f'omnibus-ruby commit ({omnibus_ruby_commit}) is not in the expected branch ({omnibus_ruby_branch}). The PR is not mergeable',
+                )
+            else:
+                print(f'Commit {omnibus_ruby_commit} was found in omnibus-ruby branch {omnibus_ruby_branch}')
```
[the remark this answers is not in the document]
Comment (chouquette) [reply tasks/release.py]: I think that's a very good point! I pushed a new commit to reduce the code duplication (which also handles the clone part, but still heavily based on the sample above)
Thanks!

**Review by chouetz — APPROVED** [review]
As long as tests are passing and you restore the values on `release.json` LGTM, thanks for the modification

**Comment by chouquette** [comment]
> As long as tests are passing and you restore the values on `release.json` LGTM, thanks for the modification

I don't think we want to revert the release.json changes, since the check would start failing as soon as this is merged

**Pull request merged**